mono/mini/interp/transform.c

```diff
@@ -3006,8 +3006,7 @@ generate (MonoMethod *method, MonoMethodHeader *header, InterpMethod *rtm, unsig
 		case CEE_THROW:
 			CHECK_STACK (td, 1);
 			SIMPLE_OP (td, MINT_THROW);
-			--td->sp;
-			generating_code = 0;
+			td->sp = td->stack;
```

Comment:
I agree with the stack reset, but I think removing `generating_code = 0` is wrong. As far as I remember it's valid according to the spec that unreachable code is allowed to be "invalid". `generating_code = 0` causes the parser to fall through the byte stream until we hit another basic block start (as it is determined by `transform.c` earlier in the process).

Reply:
of course you "fixed" the one IL test that is testing this odd behaviour 🙂

Reply:
@lewurm Good point. Removed generating_code anyway since it is broken and replaced it with the simple basic block pass that we use with the JIT.